### ISSUE TITLE
建议加个UltralightDigitalHuman版本

### ISSUE THREAD

**Issue by cacard** [lipku/LiveTalking#317]
UltralightDigitalHuman的口型效果还是很不错的，比你现在用的几个方案的口型效果都要好，建议作者加上。

**Comment by lijihua2017** [comment]
支持

**Comment by kangkang721** [comment]
大佬怎么使用，可以指点一下么？

**Comment by Surrin1999** [comment]
支持

**Comment by Chengyang852** [comment]
支持

**Comment by lijihua2017** [comment]
> UltralightDigitalHuman的口型效果还是很不错的，比你现在用的几个方案的口型效果都要好，建议作者加上。

我个人已经完成了UltralightDigitalHuman的集成，已经提交pr到这个项目了，等待作者的回应吧

**Comment by lipku** [comment]
多谢，我周末测试一下再合并

**Comment by liang1126** [comment]
支持

**Comment by lijihua2017** [comment]
> 多谢，我周末测试一下再合并

关于静音时嘴巴不稳定问题，可以录一段静音视频做为推理，用说话的视频做训练，只要是同一个人拍视频的就行，也可以使用别的软件生成闭嘴视频做为推理视频，不要使用UltralightDigitalHuman这个项目生成的闭嘴视频做为推理

**Comment by lipku** [comment]
录制不好操作，要完全一致才行。有没有什么软件能生成闭嘴视频效果较好。

**Comment by lijihua2017** [comment]
这个我也没测试过，之前在评论区有人说用wa2lip256生成的，这个模型我也没找到，可以测试一下musetalk行不行

**Comment by cacard** [comment]
NB :)

**Comment by lijihua2017** [comment]
作者自己写了一个提取的脚本，在ultralight目录下，你看看他写的文档

**Comment by cacard** [comment]
> 作者自己写了一个提取的脚本，在ultralight目录下，你看看他写的文档

嗯，使用genaratar.py生成了三个文件：
face_imgs
full_imgs
coords.pkl

开始运行livetalking，在 reading images ... 运行到 72%，总是出现进程被killed !
【内存未占满】

~/livetalking2# sh 01_webrtc_UDH-zb01_ttsGptsovits_boeItx.sh
Using cuda for inference.
Namespace(pose='data/data_kf.json', au='data/au.csv', torso_imgs='', O=False, data_range=[0, -1], workspace='data/video', seed=0, ckpt='data/pretrained/ngp_kf.pth', num_rays=65536, cuda_ray=False, max_steps=16, num_steps=16, upsample_steps=0, update_extra_interval=16, max_ray_batch=4096, warmup_step=10000, amb_aud_loss=1, amb_eye_loss=1, unc_loss=1, lambda_amb=0.0001, fp16=False, bg_img='white', fbg=False, exp_eye=False, fix_eye=-1, smooth_eye=False, torso_shrink=0.8, color_space='srgb', preload=0, bound=1, scale=4, offset=[0, 0, 0], dt_gamma=0.00390625, min_near=0.05, density_thresh=10, density_thresh_torso=0.01, patch_size=1, init_lips=False, finetune_lips=False, smooth_lips=False, torso=False, head_ckpt='', gui=False, W=450, H=450, radius=3.35, fovy=21.24, max_spp=1, att=2, aud='', emb=False, ind_dim=4, ind_num=10000, ind_dim_torso=8, amb_dim=2, part=False, part2=False, train_camera=False, smooth_path=False, smooth_path_window=7, asr=False, asr_wav='', asr_play=False, asr_model='cpierse/wav2vec2-large-xlsr-53-esperanto', asr_save_feats=False, fps=50, l=10, m=8, r=10, fullbody=False, fullbody_img='data/fullbody/img', fullbody_width=580, fullbody_height=1080, fullbody_offset_x=0, fullbody_offset_y=0, avatar_id='zb01', bbox_shift=5, batch_size=16, customvideo_config='', tts='gpt-sovits', REF_FILE='ref_zb01.wav', REF_TEXT='在商业的广阔天地里，开启一家公司的征程', TTS_SERVER='http://172.31.64.1:9880', model='ultralight', transport='webrtc', push_url='http://localhost:1985/rtc/v1/whip/?app=live&stream=livestream', max_session=1, listenport=8010, customopt=[])
reading images...
 72%|█████████████████████████████████████████████████████████▏                     | 6251/8639 [02:48<01:15, 31.43it/s]Killed
~/livetalking2# /usr/local/lib/python3.10/multiprocessing/resource_tracker.py:224: UserWarning: resource_tracker: There appear to be 1 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '

**Comment by cacard** [comment]
> > 作者自己写了一个提取的脚本，在ultralight目录下，你看看他写的文档
> 
> 嗯，使用genaratar.py生成了三个文件： face_imgs full_imgs coords.pkl
> 
> 开始运行livetalking，在 reading images ... 运行到 72%，总是出现进程被killed ! 【内存未占满】
> 
> ~/livetalking2# sh 01_webrtc_UDH-zb01_ttsGptsovits_boeItx.sh Using cuda for inference. Namespace(pose='data/data_kf.json', au='data/au.csv', torso_imgs='', O=False, data_range=[0, -1], workspace='data/video', seed=0, ckpt='data/pretrained/ngp_kf.pth', num_rays=65536, cuda_ray=False, max_steps=16, num_steps=16, upsample_steps=0, update_extra_interval=16, max_ray_batch=4096, warmup_step=10000, amb_aud_loss=1, amb_eye_loss=1, unc_loss=1, lambda_amb=0.0001, fp16=False, bg_img='white', fbg=False, exp_eye=False, fix_eye=-1, smooth_eye=False, torso_shrink=0.8, color_space='srgb', preload=0, bound=1, scale=4, offset=[0, 0, 0], dt_gamma=0.00390625, min_near=0.05, density_thresh=10, density_thresh_torso=0.01, patch_size=1, init_lips=False, finetune_lips=False, smooth_lips=False, torso=False, head_ckpt='', gui=False, W=450, H=450, radius=3.35, fovy=21.24, max_spp=1, att=2, aud='', emb=False, ind_dim=4, ind_num=10000, ind_dim_torso=8, amb_dim=2, part=False, part2=False, train_camera=False, smooth_path=False, smooth_path_window=7, asr=False, asr_wav='', asr_play=False, asr_model='cpierse/wav2vec2-large-xlsr-53-esperanto', asr_save_feats=False, fps=50, l=10, m=8, r=10, fullbody=False, fullbody_img='data/fullbody/img', fullbody_width=580, fullbody_height=1080, fullbody_offset_x=0, fullbody_offset_y=0, avatar_id='zb01', bbox_shift=5, batch_size=16, customvideo_config='', tts='gpt-sovits', REF_FILE='ref_zb01.wav', REF_TEXT='在商业的广阔天地里，开启一家公司的征程', TTS_SERVER='http://172.31.64.1:9880', model='ultralight', transport='webrtc', push_url='http://localhost:1985/rtc/v1/whip/?app=live&stream=livestream', max_session=1, listenport=8010, customopt=[]) reading images... 72%|█████████████████████████████████████████████████████████▏ | 6251/8639 [02:48<01:15, 31.43it/s]Killed ~/livetalking2# /usr/local/lib/python3.10/multiprocessing/resource_tracker.py:224: UserWarning: resource_tracker: There appear to be 1 leaked semaphore objects to clean up at shutdown warnings.warn('resource_tracker: There appear to be %d '

-----------------------
fixed：图片太多，删除一些就OK了

**Comment by LEONPICKBOY** [comment]
1、如果使用说话视频训练模型，将data_dir复制过去，不发送说话时候嘴巴也在动
2、如果用不说话的视频训练模型，说话时候嘴巴也不动
具体应该怎么操作，让不说话时嘴巴不动

**Comment by LEONPICKBOY** [comment]
另外还有不说话嘴巴抖动的问题怎么解决

**Comment by lmc710731910** [comment]
> 有没有什么软件能生成闭嘴视频效果较好。

我现在用framepack生成静默视频
